presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotExpression.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.pinot.query;
+
+public class PinotExpression
+{
+    private final String definition;
+    private final PinotQueryGeneratorContext.Origin origin;
+
+    PinotExpression(String definition, PinotQueryGeneratorContext.Origin origin)
+    {
+        this.definition = definition;
```

Comment:
add `requireNonNull(..., "definition is null");`. Same for almost all other constructors.